diff --git a/node_modules/@strapi/admin/dist/_chunks/index-81509193.mjs b/node_modules/@strapi/admin/dist/_chunks/index-81509193.mjs
index 74d15ae..fb595b7 100644
--- a/node_modules/@strapi/admin/dist/_chunks/index-81509193.mjs
+++ b/node_modules/@strapi/admin/dist/_chunks/index-81509193.mjs
@@ -8674,7 +8674,7 @@ const DynamicComponent = ({
   dynamicComponentsByCategory,
   onAddComponent
 }) => {
-  const [isOpen, setIsOpen] = useState(true);
+  const [isOpen, setIsOpen] = useState(false);
   const { formatMessage } = useIntl();
   const { getComponentLayout } = useContentTypeLayout();
   const { modifiedData } = useCMEditViewDataManager();
