diff --git a/node_modules/@strapi/admin/dist/admin/index.mjs b/node_modules/@strapi/admin/dist/admin/index.mjs
index a8f47c4..2e02b4c 100644
--- a/node_modules/@strapi/admin/dist/admin/index.mjs
+++ b/node_modules/@strapi/admin/dist/admin/index.mjs
@@ -14,7 +14,7 @@ export { Table, useTable } from './admin/src/components/Table.mjs';
 export { ContentBox } from './admin/src/components/ContentBox.mjs';
 export { SubNav } from './admin/src/components/SubNav.mjs';
 export { GradientBadge } from './admin/src/components/GradientBadge.mjs';
-export { tours } from './admin/src/components/GuidedTour/Tours.mjs';
+export { tours as unstable_tours } from './admin/src/components/GuidedTour/Tours.mjs';
 export { useGuidedTour } from './admin/src/components/GuidedTour/Context.mjs';
 export { GUIDED_TOUR_REQUIRED_ACTIONS } from './admin/src/components/GuidedTour/utils/constants.mjs';
 export { HEIGHT_TOP_NAVIGATION, HEIGHT_TOP_NAVIGATION_MEDIUM, RESPONSIVE_DEFAULT_SPACING, WIDTH_SIDE_NAVIGATION } from './admin/src/constants/theme.mjs';
