diff --git a/node_modules/@strapi/plugin-users-permissions/server/content-types/user/index.js b/node_modules/@strapi/plugin-users-permissions/server/content-types/user/index.js
index d31322a..2f8344c 100644
--- a/node_modules/@strapi/plugin-users-permissions/server/content-types/user/index.js
+++ b/node_modules/@strapi/plugin-users-permissions/server/content-types/user/index.js
@@ -15,6 +15,11 @@ module.exports = {
     draftAndPublish: false,
     timestamps: true,
   },
+  pluginOptions: {
+    "content-manager": {
+      "visible": false
+    }
+  },
   attributes: {
     username: {
       type: 'string',
