diff --git a/node_modules/@strapi/admin/dist/_chunks/index-uCIlbOdV.mjs b/node_modules/@strapi/admin/dist/_chunks/index-uCIlbOdV.mjs
index 95a5860..9f3bdd7 100644
--- a/node_modules/@strapi/admin/dist/_chunks/index-uCIlbOdV.mjs
+++ b/node_modules/@strapi/admin/dist/_chunks/index-uCIlbOdV.mjs
@@ -13537,7 +13537,7 @@ const DynamicComponent = ({
   dynamicComponentsByCategory = {},
   onAddComponent
 }) => {
-  const [isOpen, setIsOpen] = React.useState(true);
+  const [isOpen, setIsOpen] = React.useState(false);
   const { formatMessage } = useIntl();
   const { getComponentLayout } = useContentTypeLayout();
   const { modifiedData } = useCMEditViewDataManager();
