diff --git a/node_modules/@strapi/admin/admin/src/content-manager/components/DynamicZone/components/DynamicComponent.js b/node_modules/@strapi/admin/admin/src/content-manager/components/DynamicZone/components/DynamicComponent.js
index f714b3e..0512edf 100644
--- a/node_modules/@strapi/admin/admin/src/content-manager/components/DynamicZone/components/DynamicComponent.js
+++ b/node_modules/@strapi/admin/admin/src/content-manager/components/DynamicZone/components/DynamicComponent.js
@@ -37,7 +37,7 @@ export const DynamicComponent = ({
   dynamicComponentsByCategory,
   onAddComponent,
 }) => {
-  const [isOpen, setIsOpen] = useState(true);
+  const [isOpen, setIsOpen] = useState(false);
   const { formatMessage } = useIntl();
   const { getComponentLayout } = useContentTypeLayout();
   const { modifiedData } = useCMEditViewDataManager();
